src/SFA.DAS.EmployerAccounts/Queries/GetReservations/GetReservationsRequestHandler.cs

```diff
@@ -42,10 +42,24 @@ public async Task<GetReservationsResponse> Handle(GetReservationsRequest message
 
             try
             {
+                var task = _service.Get(accountId);
+                if (await Task.WhenAny(task, Task.Delay(message.TimeOut)) == task)
+                {
+                    await task;
+                }
+                return new GetReservationsResponse
+                {
+                    Reservations = task.Result
+                };
+            }
+            catch (TimeoutException ex)
```

Comment:
This handler seems to be the only one to specifically handle a timeout exception rather than utilizing the else{} block?